sse2neon.h

```diff
@@ -7748,6 +7748,80 @@ FORCE_INLINE __m128i _mm_minpos_epu16(__m128i a)
     return dst;
 }
 
+// Compute the sum of absolute differences (SADs) of quadruplets of unsigned
+// 8-bit integers in a compared to those in b, and store the 16-bit results in
+// dst. Eight SADs are performed using one quadruplet from b and eight
+// quadruplets from a. One quadruplet is selected from b starting at on the
+// offset specified in imm8. Eight quadruplets are formed from sequential 8-bit
+// integers selected from a starting at the offset specified in imm8.
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_mpsadbw_epu8
+FORCE_INLINE __m128i _mm_mpsadbw_epu8(__m128i a, __m128i b, const int imm)
+{
+    uint8x16_t _a, _b;
+
+    switch (imm & 0x4) {
```

Comment:
Is it possible to have uninitialized `_a`? e.g., when `(imm & 0x4) == 3`

Reply:
But I think the result of `imm&0x4` will be either 0 or 4, therefore, other values are impossible

Reply:
Would you add some compiler hints such as `__builtin_unreachable` there?

Reply:
I have added `__builtin_unreachable()` to the `default` two switch-case